app/src/main/java/org/luxoft/sdl_core/MainActivity.java

```diff
@@ -100,6 +94,50 @@ public void onClick(View view) {
             }
         });
 
+        start_bt_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (!isBleSupported() || !isBluetoothPermissionGranted()) {
+                    start_bt_button.setEnabled(false);
```

Comment:
@ychernysheva I think launcher should disable these buttons if BLE is not supported or permissions were not granted. Check inside the button itself is not needed. User simply should not be able to click it in that case.